Makefile

```diff
@@ -191,6 +189,9 @@ deploy-mutation: patch-image
 
 # Deploy controller in the configured Kubernetes cluster in ~/.kube/config
 deploy: patch-image manifests
+ifeq ($(ENABLE_EXTERNAL_DATA),true)
```

Comment:
Do we want to create a separate target like `deploy-mutation` and move this there?